api/app/controllers/spree/api/v2/platform/payment_methods_controller.rb

```diff
@@ -0,0 +1,29 @@
+module Spree
+  module Api
+    module V2
+      module Platform
+        class PaymentMethodsController < ResourceController
+          private
+
+          def model_class
+            Spree::PaymentMethod
+          end
+
+          def spree_permitted_attributes
+            preferred_attributes = []
+
+            if action_name == 'update'
+              resource.defined_preferences.each do |preference|
```

Comment:
@MatthewKennedy we could move this code to the parent `spree_permitted_attributes` method as it will be used across many models

Reply:
Moving `resource.defined_preferences` to `spree_permitted_attributes` turned out to be quite messy, it is doable, but:

We would have to write a conditional in the `resource` method to create a new instance, then we can check if this new instance has any preferences, as it stands resource. is not available until after the permitted attributes have been set for the create action.

I think this might be better done in its own PR.

Reply:
@MatthewKennedy let's drop the preferences work and leave as it is, we're not going to support it anymore soon

Reply:
@MatthewKennedy I think we can even drop it completely not to let people use it for one version and later remove it